libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt

```diff
@@ -328,8 +328,10 @@ internal object MapboxRouteLineUtils {
         routeLineExpressionData: List<RouteLineExpressionData>,
         lineBaseColor: Int
     ): List<RouteLineExpressionData> {
-        val filteredItems = routeLineExpressionData
-            .filter { it.offset > distanceOffset }.distinctBy { it.offset }
+        val filteredItems = routeLineExpressionData.filterIndexed { index, expressionData ->
+            expressionData.offset > distanceOffset &&
+                expressionData.offset != routeLineExpressionData.getOrNull(index + 1)?.offset
+        }
```

Comment:
I noticed you removed the .distinctBy. Can you be absolutely sure there will be no duplicates in the offset with multi-leg routes?  This was added in response to duplicates being produced with multi-leg routes. A duplicate value will produce an invalid `Expression` and crash the app.

Reply:
Yes, we're iterating over the offsets and filtering out the duplicates. The code works nearly the same way as `distinctBy` but retains the last occurrence in a chain of duplicates, instead of the first one.

The return would only differ if we had duplicate offset at different points in the geometry (not adjacent to each other) but I don't know how that could happen.

Is there a specific test case you have in mind that we could run against it?

Reply:
I remember this was added because of multi-leg routes.  I think the last point in one leg was the same as the first point in the next leg.  If that condition is accounted for in a different way by the granular distances calc. then we should be OK.

Reply:
Yes, this is accounted for in https://github.com/mapbox/mapbox-navigation-android/blob/fca00080cb6b7b802515066e42dcd1b9500eab95/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt#L949-L952 and https://github.com/mapbox/mapbox-navigation-android/blob/fca00080cb6b7b802515066e42dcd1b9500eab95/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt#L963-L965